### PR TITLE
Support bitbucket [Building upon #328]

### PR DESCRIPTION
**Pull request by tvytlx**
### Description
Fix all requests for improvements in #328.

### Proposal
Support bitbucket

### CODE REVIEW

File: src/adapters/bitbucket.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      // Get username of logged-in user.
+      const currentUser = JSON.parse($('body').attr('data-current-user'))
+
+      // Or get username by spliting token.
+      if (opts.token.includes(':')) {
+        const result = opts.token.split(':')
+        const username =  result[0], token = result[1]
+        cfg.headers = { Authorization: 'Basic ' + btoa(username + ':' + token) }
+      }
+      else if (currentUser.username) {
```
Comment (buunguyen) [inline src/adapters/bitbucket.js]: Can `currentUser` be null here if the BB DOM has been changed? Maybe simplify logic:

```
let username = null
if (opts.token.includes(':')) {
   username = ...
}
else {
  const currentUser = JSON.parse($('body').attr('data-current-user'))
  if (!currentUser || !currentUser.username) {
     return cb(error_object_prompt_for_username_and_token)
  }
  username = ...
}
cfg.headers = ...
```

File: src/adapters/bitbucket.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    if (~BB_RESERVED_USER_NAMES.indexOf(username) ||
+        ~BB_RESERVED_REPO_NAMES.indexOf(reponame) ||
+        ~BB_RESERVED_TYPES.indexOf(type)) {
+      return cb()
+    }
+
+    // Skip non-code page unless showInNonCodePage is true
+    // with Bitbucket /username/repo is non-code page
+    if (!showInNonCodePage &&
+      (!type || (tyep && type !== 'src'))) {
```
Comment (buunguyen) [inline src/adapters/bitbucket.js]: typo `tyep`

File: src/adapters/bitbucket.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      }
+    }
+
+    $.ajax(cfg)
+      .done((data) => cb(null, data))
+      .fail((jqXHR) => {
+        if (opts.token && !opts.token.includes(':')) {
+          cb({
+            error: 'Error: Invalid token',
+            message: `Cannot get your username from current page,
```
Comment (buunguyen) [inline src/adapters/bitbucket.js]: 'Cannot retrieve your user name from the current page. Please update the token setting to prepend your user name to the token, separated by a colon, i.e. USERNAME:TOKEN'.

**Comment by buunguyen** [comment]
Thanks. I've tested on BitBucket and seems to work great. 
Would you mind update README.md with info to configure BitBucket?

**Comment by tvytlx** [comment]
Ok, I try to add a commit.

**Comment by tvytlx** [comment]
@buunguyen Is this modification reasonable?

**Comment by buunguyen** [comment]
Great work, thank you!

**Pull request merged**